### PR TITLE
acceptGroupV4Invite

### PR DESCRIPTION
**Pull request by Darker935**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -507,6 +508,8 @@ declare namespace WAWebJS {
         location: Location,
         /** List of vCards contained in the message */
         vCards: string[],
+        /** Invite v4 info */
+        invite: object
```
Comment (pedroslopez) [inline index.d.ts:512]: We should offer better typings for this object and what it contains

File: src/Client.js [hunk 2]
```diff
@@ -584,6 +584,20 @@ class Client extends EventEmitter {
...
     }
 
+    /**
+     * Accepts a private invitation to join a group
+     * @param {object} inviteInfo Invite V4 Info
+     * @returns {Promise<Object>}
+     */
+     async acceptGroupV4Invite(inviteInfo) {
+        if(!inviteInfo.inviteCodeV4) return 'Invalid invite code'
+        if (inviteInfo.inviteCodeExpV4 == 0) return 'Expired invite code'
```
Comment (pedroslopez) [inline src/Client.js:594]: We should `throw` here instead of returning to indicate something went wrong

**Review by pedroslopez — CHANGES_REQUESTED** [review]
Cool!

This makes me think of a couple questions:
- Do the old invites still work?
- How do these differ from the old invites?
- Could something be done to make the existing `acceptInvite` function work with the new format?

Also, it seems like typings are missing for the new `acceptGroupV4Invite` function.

**Comment by JoseHM8A** [comment]
> 
> 
> Cool!
> 
> This makes me think of a couple questions:
> 
>     * Do the old invites still work?
> 
>     * How do these differ from the old invites?
> 
>     * Could something be done to make the existing `acceptInvite` function work with the new format?
> 
> 
> Also, it seems like typings are missing for the new `acceptGroupV4Invite` function.

1° - Yes, old invites are links, that anyone can join using them
2° - The InviteV4 is a private invite that only the user can use (it is received when the user does not allow to be added to groups by not contacts)
3° - The new method requires additional parameters, like expiration time, invite code, the user who received and who sent, that are in the message data

File: index.d.ts (hunk 1 above)
Comment (JoseHM8A) [reply index.d.ts]: Done (i think), this is my first PR 😅

**Review by pedroslopez — APPROVED** [review]

**Comment by pedroslopez** [comment]
Thanks! Tested this and it worked great. Just made a few changes to the typings.

**Pull request merged**